backend/x86_binary_emitter.ml

```diff
@@ -1065,6 +1126,26 @@ let emit_j b loc condition dst =
         b loc symbol
   | _ -> assert false
 
+let imm8_of_float_condition = function
+  | EQf -> 0x00
+  | LTf -> 0x01
+  | LEf -> 0x02
+  | UNORDf -> 0x03
+  | NEQf -> 0x04
+  | NLTf -> 0x05
+  | NLEf -> 0x06
+  | ORDf -> 0x07
+
+let emit_cmpsd b ~condition ~dst ~src =
+  match (dst, src) with
+  | (Regf reg, ((Regf _ | Mem _ | Mem64_RIP _) as rm)) ->
+    (* CMPSD xmm1, xmm2/m64, imm8 *)
+    let condition = imm8_of_float_condition condition in
+    buf_int8 b 0xF2;
```

Comment:
`0x66`, `0xF2` and `0xF3` are mandatory prefixes and the rex prefix must be inserted after those when the instructions has them so I prefer to stay consistent and always output those prefixes separately.

It might be the case that the original emitter isn't always consistent wrt to those but I tried to be in the support for the new instructions.